[hxb] Add actual positions to unbound type parameter warnings

kLabz opened the pull request:
* [x] Removed deduplication now that we have meaningful positions; though `cf_expr` vs `cf_unoptimized_expr` duplicates are still silly..
* [x] Some unhandled cases (added `TODO`s there)

See https://github.com/HaxeFoundation/haxe/pull/11628 for some easy to reproduce examples of unbound type parameter warnings

kLabz committed:
[WIP] Add actual positions to unbound type parameter warnings

kLabz commented:
@Simn should I try to include that in preview?

Simn commented:
I have no idea how that effect stuff works but it is useful and seems harmless, so why not!

kLabz commented:
I think this can be useful on several parts of our codebase.

It's basically an exception that we can resume from: we raise a specific exception, and it can be caught. When catching it, we provide a value to the call site through a continuation function.

Need to be cautious with it, though, or at least add the unhandled effect exception in our catch-all

kLabz committed:
Catch unbound ttp effects in write_module

(avoids unhandled effect issues for cases I missed)

kLabz committed:
Only report each (msg,p) warning once

kLabz committed:
Handle unbound type warning positions for abstract this/from/to and typedef

Other instances should be fine, and would be caught by already handled cases or write_module itself
